src/main/java/edu/ucsb/ucsbcslas/controllers/TutorAssignmentController.java

```diff
@@ -221,55 +230,114 @@
+
+        String empty = "";
+        empty = new String(csv.getInputStream().readAllBytes());
+        if ((authControllerAdvice.getIsAdmin(authorization)) && ("" != empty)) {
+            try {
+                List<TutorAssignmentModel> uploadedTutorAssignments = csvToObjectService.parse(reader,
+                        TutorAssignmentModel.class);
+                List<TutorAssignment> tutorAssignmentList = new ArrayList<TutorAssignment>();
+
+                for (TutorAssignmentModel i : uploadedTutorAssignments) {
+                    Course currentCourse;
+                    Tutor currentTutor;
+                    TutorAssignment currentTutorAssignment;
+
+                    List<Course> courseList = courseRepository.findByNameAndQuarterAndInstructorEmail(i.getCourseName(),
+                            i.getQuarter(), i.getInstructorEmail());
+                    if (courseList.isEmpty()) {
+                        currentCourse = new Course(i.getCourseName(), i.getQuarter(), i.getInstructorFirstName(),
+                                i.getInstructorLastName(), i.getInstructorEmail());
+                        courseRepository.save(currentCourse);
+                    } else {
+                        currentCourse = courseList.get(0);
+                    }
+                    Optional<Tutor> tutorList = tutorRepository.findByEmail(i.getTutorEmail());
+                    if (tutorList.isEmpty()) {
+                        currentTutor = new Tutor(i.getTutorFirstName(), i.getTutorLastName(), i.getTutorEmail());
+                        tutorRepository.save(currentTutor);
+                    } else {
+                        currentTutor = tutorList.get();
+                    }
+                    tutorAssignmentList = tutorAssignmentRepository.findAllByTutor(currentTutor);
+                    if (tutorAssignmentList.isEmpty()) {
+                        currentTutorAssignment = new TutorAssignment(currentCourse, currentTutor,
+                                i.getAssignmentType());
+                        tutorAssignmentRepository.save(currentTutorAssignment);
+                    }
+                }
 
+                List<TutorAssignment> savedTutorAssignments = (List<TutorAssignment>) tutorAssignmentRepository
+                        .saveAll(tutorAssignmentList);
+                String body = mapper.writeValueAsString(savedTutorAssignments);
+                System.out.print(savedTutorAssignments.toString());
+                return ResponseEntity.ok().body(body);
 
-      newAssignment.setId(id);
-      tutorAssignmentRepository.save(newAssignment);
-      String body = mapper.writeValueAsString(newAssignment);
-      return ResponseEntity.ok().body(body);
+            } catch (RuntimeException e) {
+                logger.error("Error Uploading CSV", e);
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Malformed CSV", e);
+            }
+        } else {
+            return null;
```

Comment:
I think instead of returning null here, we want to return a BAD_REQUEST similar to line 336.